src/main/java/com/amazon/jenkins/ec2fleet/EC2FleetCloud.java

```diff
@@ -209,9 +209,13 @@ public static void log(final Logger logger, final Level level,
         if (stats.getNumDesired() >= maxAllowed || !"active".equals(stats.getState()))
             return Collections.emptyList();
 
-	// Presumably ceil() would also be correct but I think round() works best for
-	// scenarios when numExecutors is of reasonable size.
-        int weightedExcessWorkload = Math.round((float)excessWorkload / this.numExecutors);
+        // Ensure we don't divide by 0, below
+        if (this.numExecutors == 0)
+            return Collections.emptyList();
+
+        // Calculate the ceiling, without having to work with doubles from Math.ceil
+        // https://stackoverflow.com/a/21830188/877024
+        int weightedExcessWorkload = (excessWorkload - 1) / this.numExecutors + 1;
```

Comment:
I don't really buy the current math.
```
>>> 494./21
23.523809523809526
>>> 493./21
23.476190476190474
>>> (494 - 1) / (21 + 1)   # this PR
22
```

Why not keep the rounding behavior the same?

```java
int weightedExcessWorkload = ((excessWorkload * 2) + this.numExecutors) / (this.numExecutors * 2);
```
```
>>> (493*2 + 21)/42        # suggested rounding
23
>>> (494*2 + 21)/42        # suggested rounding
24
```

If you still want the ceiling, you can do the following, which is also identified by your SO link:
```java
int weightedExcessWorkload = (excessWorkload + this.numExecutors - 1) / this.numExecutors;
```
```
>>> 483./21
23.0
>>> 484./21
23.047619047619047
>>> 503./21
23.952380952380953
>>> 504./21
24.0
>>> (483 + 21 - 1)/21
23
>>> (484 + 21 - 1)/21
24
>>> (503 + 21 - 1)/21
24
>>> (504 + 21 - 1)/21
24
```
These functions are a transformation of the common way of building rounding and ceiling functions out of a floor function:
```
round(x) = floor(x + .5)
ceil(x) = floor(x + 1 - ε)
```
We modify these for int division:
```
round(a / b) = floor((a / b) + .5) = floor((a / b) + (.5b / b)) = floor((a + .5b) / b) = floor((2a + b) / 2b)
ceil(a / b) = floor((a / b) + 1 - ε)+ floor((a / b) + (b / b) - ε) = floor(((a + b) / b) - ε)
```
For the `ceil` function, we break at the limit (no remainder), so we need to subtract some epsilon value `ε`. With integer division `ε` can be `1 / b` because it's guaranteed to be the smallest unit that can affect the calculation.
So we get
```
ceil(a / b) = floor((a + b - 1) / b)
```

Reply:
Thanks for the reply!

> Why not keep the rounding behavior the same?

Because with my example, where excessWorkload is 1 and numExecutors is 3:

```
In [3]: ((1 * 2) + 3) / (3 * 2)
Out[3]: 0
```

Which means that you need 2 jobs in queue before you trigger the spot fleet to scale up -- the whole point of this PR is that we have times on the weekend where we have a target of 0 in the spot fleet and only 1 job comes into the queue.  It has a timeout on it, and eventually times out before another job enters the queue.

I agree with your example where excessWorkload is 494 and numExecutors is 21.  So, going with:

```
int weightedExcessWorkload = (excessWorkload + this.numExecutors - 1) / this.numExecutors;
```

Is fine by me.  I'll adjust accordingly.  If you look at the SO answer I linked the current implementation is the variation of option 2, the reason I picked it over the above is because it's called out as:

> To reduce the chance of overflow

But like I said, either is fine by me so long as my spot fleet will scale up with only 1 job in queue :)

Changes incoming!